chapters/ch55.md

```diff
@@ -1,78 +1,82 @@
 
 > 1. Try manually modifying A (detecting cars)’s output to be a "perfect" output (e.g., manually go in and tell it where the other cars are). Run the rest of the pipeline B, C as before, but allow C (plan path) to use A’s now perfect output. If the algorithm now plans a much better path for the car, then this shows that, if only A had given a better output, the overall algorithm’s output would have been better; Thus, you can attribute this error to component A. Otherwise, go on to Step 2.
 
---> _replace THIS LINE by your translation for the above line_
+1. Thử điều chỉnh thủ công kết quả đầu ra của thành phần A (phát hiện xe hơi) sao cho "hoàn hảo" (ví dụ, cho xe biết vị trí của những chiếc xe khác). Sau đó chạy thuật toán trên  pipeline gồm có B và C, sủ dụng đầu ra "hoàn hảo" của A đã được thao tác thủ công trước khi đưa vào thành phần C (hoạch định đường đi cho xe). Nếu thuật toán hoạch định đường đi cho xe tốt hơn, điều đó cho thấy rằng, kết quả cuối cùng của thuật toán tự lái sẽ được cải thiện khi và chi khi thành phần A cho kết quả đầu ra tốt hơn.  Như vậy, bạn có thể quy lỗi cho A. Ngược lại, ta tiếp tục bước 2. 
 
 > 2. Try manually modifying B (detect pedestrian)’s output to be the "perfect" output for B. If the algorithm now gives a correct output, then attribute the error to component B. Otherwise, go on to Step 3.
 
---> _replace THIS LINE by your translation for the above line_
+Tương tự, ta thử điều chỉnh thủ công kết quả đầu ra của thành phần B (phát hiện người đi bộ) sao cho "hoàn hảo". Nếu thuật toán cho ra kết quả lái xe tốt, ta có thể quy lỗi cho B. Ngược lại, ta tiếp tục ở bước 3.
 
 > 3. Attribute the error to component C.
 
---> _replace THIS LINE by your translation for the above line_
+Sau khi đã kiểm chứng nguyên nhân phát sinh lỗi không phải do A và B gây ra, ta quy lỗi cho thành phần C. 
 
 > The components of an ML pipeline should be ordered according to a Directed Acyclic Graph (DAG), meaning that you should be able to compute them in some fixed left-to-right order, and later components should depend only on earlier components’ outputs. So long as the mapping of the components to the A->B->C order follows the DAG ordering, then the error analysis will be fine. You might get slightly different results if you swap A and B:
 
---> _replace THIS LINE by your translation for the above line_
+Các thành phần của một mô hình học máy dạng pipeline nên được tổ chức theo lý thuyết đồ thị định hướng không tuần hoàn (DAG), nghĩa là bạn có thể tính toán chúng theo thứ tự cố định từ trái sang phải, và các thành phần sau sẽ phụ thuộc chỉ vào kết quả đầu ra của các thành phần trước đó. Miễn là việc xâu chuỗi các thành phần theo thứ tự A->B->C tuân thủ theo quy tắc đồ thị định hướng không tuần hoàn (DAG), việc phân tích lỗi sẽ tốt. Bạn có thể nhận được các kết quả hơi khác nhau nếu hoán chuyển vị trí của A và B cho nhau như sau:
```

Comment:
* "của một mô hình học máy dạng pipeline" ->"của một pipeline học máy"
* "~tổ chức theo lý thuyết~ xắp xếp thành đồ thị định hướng không tuần hoàn (DAG)"
* "từ trái sang phải _nào đó_" . Mình nghĩ từ "some" quan trọng không nên lược bỏ.
* "sẽ phụ thuộc chỉ vào kết quả đầu ra" -> "chỉ nên phụ thuộc vào đầu ra"

Reply:
> "của một mô hình học máy dạng pipeline" ->"của một pipeline học máy"

Mình đồng ý với cách dịch "của một pipeline học máy" tuy nhiên mình lại thấy cách dịch "của một mô hình học máy dạng pipeline" thì dễ cho mình hình dung hơn, đó là mô hình học máy xây dựng theo kiến trúc pipeline. Mình không biết với các bạn đọc tương lai của cuốn sách này, thì ta nên dùng cách nào cho phù hợp?

> "xắp xếp thành đồ thị định hướng không tuần hoàn (DAG)"
Mình e rằng chữ "thành" ở đây có thể làm cho cụm từ "according to" sẽ không được truyền tải rõ ràng. Ở đây, mình hiểu các thành phần của "an ML pinpeline" nên được  sắp xếp theo  (quy tắc/quy luật) đồ thị định hướng không tuần hoàn. 

Do đó, tạm thời mình sẽ kết hợp phần bạn góp ý và phần mình hiểu thành "Các thành phần của một mô hình học máy dạng pipeline nên được sắp xếp theo đồ thị định hướng không tuần hoàn (DAG)".

Reply:
> "từ trái sang phải nào đó" . Mình nghĩ từ "some" quan trọng không nên lược bỏ.

Mình đã chỉnh sửa theo gợi ý.

Reply:
> later components should depend only on earlier components’ outputs

> "sẽ phụ thuộc chỉ vào kết quả đầu ra" -> "chỉ nên phụ thuộc vào đầu ra"

Mình  đã sữa "sẽ" --> "nên". Riêng từ "chỉ" mình nghĩ để gần với danh từ  sẽ phù hợp hơn vì "only depend" và "depend only" mang nghĩa khác nhau.

Reply:
Theo mình thì "phụ thuộc chỉ vào" hay "chỉ phụ thuộc vào" không khác nhau nhiều về nghĩa. Tuy nhiên cụm thứ nhất nghe không được tự nhiên, mình chưa gặp bao giờ.

Reply:
Trong tiếng Anh thì khi viết "only depend + Noun" và "depend only + Noun"  khác nhau rõ ràng. Tuy nhiên, khi dịch sang tiếng Việt đôi khi không mang rõ nghĩa như tiếng Anh. Mình cũng chịu không biết nên diễn đạt thế nào trong tiếng Việt cho tốt hơn, nên mình sẽ theo gợi ý của 2 bạn. Mình cập nhật lại "chỉ nên phụ thuộc vào đầu ra".